js/preview/src/index.ts

```diff
@@ -156,14 +156,15 @@ export function examine_module(
 		.trim()
 		.split("\n")
 		.map((line) => {
-			const [name, template_dir, frontend_dir, component_class_id] =
-				line.split("~|~|~|~");
-
-			return {
-				name: name.trim(),
-				template_dir: template_dir.trim(),
-				frontend_dir: frontend_dir.trim(),
-				component_class_id: component_class_id.trim()
-			};
-		});
+			if (line !== "~|~|~|~NOT INSTALLED~|~|~|~") {
```

Comment:
We might want to throw an error for this case. But it might actually be safer to check that the next 4 argument are always present instead of this because if there is some kind of exception thrown from the python file we won't catch it with either of these branches.

Reply:
I don't think we should throw an error. Because maybe a component that's not used in the demo you're running is not installed. Don't think we should stop the dev mode launch in that case.

But let me rework to check for all four variables being present!